### ISSUE TITLE
Provide a Redhat / CentOS aarch64 build

### ISSUE THREAD

**Issue by fdr** [crystal-lang/crystal#9833]
Noticed this was missing. aarch64 servers are a lot more common than they used to be on account of AWS's newish instance lines.

```
# cat crystal.repo 
[crystal]
name = Crystal
baseurl = https://dist.crystal-lang.org/rpm/

# yum install crystal
Crystal                                                                                                                                                                                                       1.1 MB/s | 909 kB     00:00    
Extra Packages for Enterprise Linux Modular 8 - aarch64                                                                                                                                                       164 kB/s |  97 kB     00:00    
Extra Packages for Enterprise Linux 8 - aarch64                                                                                                                                                               3.7 MB/s | 7.9 MB     00:02    
Error: 
 Problem: cannot install the best candidate for the job
  - package crystal-0.35.1-1.i386 does not have a compatible architecture
```

**Comment by fdr** [comment]
I notice this spec file found in the wild on fedora copr uses the bare crystal binary releases for bootstrapping, so making those available may help proliferate the aarch64 support as well:

https://copr-dist-git.fedorainfracloud.org/cgit/zawertun/crystal/crystal.git/tree/crystal.spec?h=f33 via https://copr.fedorainfracloud.org/coprs/zawertun/crystal/

**Comment by fdr** [comment]
Bintray releases also short on aarch64. https://dl.bintray.com/crystal/rpm/

**Comment by mistergibson** [comment]
While you're at it --> build an official aarch64 release for Ubuntu.

**Comment by straight-shoota** [comment]
Once aarch64 is fully supported we will provide binary releases and packages for all these platforms. But we're not there yet.

However, it would be totally feasible to provide unofficial community builds if you already want to start working with it.
There is https://hub.docker.com/r/danielworrall/crystal for example.

**Comment by kostya** [comment]
seems main blocking issue for linux-aarch64 was closed https://github.com/crystal-lang/crystal/issues/9533, is there any else?

**Comment by straight-shoota** [comment]
No, aarch64 should be good. We just haven't managed to build distribution packages yet. The problem is that our build pipeline runs on circleci which doesn't have aarch64 hosts. Work has started to refactor the CI setup, but it's not ready yet.

**Comment by draeath** [comment]
Any luck on this? I was trying to run some software that requires 1.8.2 and found there's no arm64/aarch64 builds - and given that one requires crystal to build crystal...

**Comment by straight-shoota** [comment]
There are still no official images for aarch64.
But you can use the community images from https://hub.docker.com/r/84codes/crystal

**Comment by will** [comment]
@draeath it might be too much trouble for you than it's worth depending on your situation but aarch64-linux crystal support is packaged in nix. It's not a solution to this issue, but just letting you know in case it can unblock you

```
[will@nixos:~]$ nix shell nixpkgs#crystal

[will@nixos:~]$ crystal --version
Crystal 1.9.2 (1980-01-01)

LLVM: 15.0.7
Default target: aarch64-unknown-linux-gnu
```

**Comment by draeath** [comment]
That's actually very helpful, I can dig around into how the nix package is built and might have some luck from that angle.

Thank you! Likewise that community container, the layers data (or their source repo if available) may also help to show how it was built.

**Comment by straight-shoota** [comment]
Starting with [1.18.2](https://github.com/crystal-lang/crystal/releases/tag/1.18.2) we're providing tarballs with release builds for aarch64.
Respective docker images should follow soon.

RPM and DEB packages for aarch64 are also enabled, but unfortunately we're a bit limited on platform support by our package build service. Currently, we can only provide aarch64 packages for openSUSE 16.0 and Fedora repositories.

As mentioned before, there are other installation options with better aarch64 coverage than upstream packages. Please refer to https://crystal-lang.org/install/#linux
